File: edgevdb/python/edgevdb/object_store.py
"""
EdgeVDB Python object_store module — re-exported from __init__.py.
Provides convenience methods for the relational object store.
"""

from edgevdb import EdgeVDB

__all__ = ["ObjectStoreHelper"]


class ObjectStoreHelper:
    """Convenience wrapper for object store operations."""

    def __init__(self, db: EdgeVDB):
        self._db = db

    def put(self, type_name: str, properties: dict) -> int:
        return self._db.put_object(type_name, properties)

    def get(self, object_id: int) -> dict:
        return self._db.get_object(object_id)

    def remove(self, object_id: int):
        self._db.remove_object(object_id)

    def add_relation(self, name: str, from_id: int, to_id: int):
        self._db.add_relation(name, from_id, to_id)

File: ios/Sources/EdgeVDB/ObjectStore.swift
import Foundation

/// ObjectStore — Swift wrapper for the relational object database.
public final class ObjectStore {
    private weak var db: EdgeVDB?

    internal init(db: EdgeVDB) {
        self.db = db
    }

    public func put(type: String, properties: [String: Any]) throws -> UInt64 {
        guard let db = db else { throw EdgeVDBError.nullHandle }
        return try db.putObject(type: type, properties: properties)
    }

    public func get(_ id: UInt64) throws -> [String: Any]? {
        guard let db = db else { throw EdgeVDBError.nullHandle }
        return try db.getObject(id)
    }

    public func remove(_ id: UInt64) throws {
        guard let db = db else { throw EdgeVDBError.nullHandle }
        try db.removeObject(id)
    }

    public func addRelation(_ name: String, from: UInt64, to: UInt64) throws {
        guard let db = db else { throw EdgeVDBError.nullHandle }
        try db.addRelation(name, from: from, to: to)
    }
}
